### ISSUE TITLE
BeanItem for immutable classes

### ISSUE THREAD

**Issue by henrikerola** [henrikerola/scaladin#35]

**Comment by fancellu** [comment]
Yes, that would be very nice. Currently have to put @BeanProperty on things 
Would be also cool if it could handle editing immutable classes. Yes, I know. 
I mean doing copy(field=newval) on field edit, Then handing over this final class for application specific handling on commit.
